qa/tasks/cephfs/test_admin.py

```diff
@@ -2695,49 +2695,50 @@ def test_with_health_warn_trim(self):
 
     def _get_unhealthy_mds_id(self, health_warn):
         '''
-        Return MDS ID for which health warning in "health_warn" has been
-        generated.
+        Returns number of MDSs for which health warning in "health_warn" has been
+        generated and the first MDS ID for which warning is generated
         '''
         health_report = json.loads(self.get_ceph_cmd_stdout('health detail '
-                                                            '--format json'))
+                                                            '--format json-pretty'))
         # variable "msg" should hold string something like this -
         # 'mds.b(mds.0): Behind on trimming (865/10) max_segments: 10,
         # num_segments: 86
+        count = health_report['checks'][health_warn]['summary']['count']
         msg = health_report['checks'][health_warn]['detail'][0]['message']
         mds_id = msg.split('(')[0]
         mds_id = mds_id.replace('mds.', '')
-        return mds_id
+        return count, mds_id
```

Comment:
Returning `count` seems unnecessary, we return a tuple of MDS IDs for which health warning is raised. And then getting length of tuple would do the job. Therefore, collecting and returning count is redundant.

Reply:
Isn't the function always returning the single mds_id ?
The count was introduced to catch the bug if any. It's used in testcode to assert that only mds is expected to raise the warning.

Reply:
Okay, the count is to ensure that warning was raised for only one MDS. It makes sense now.